crates/sui-benchmark/tests/simtest.rs

```diff
@@ -249,10 +257,118 @@ mod test {
         assert_eq!(expected_checkpoint, pruned);
     }
 
+    #[sim_test(config = "test_config()")]
+    async fn test_testnet_upgrade_compatibility() {
+        // This test is intended to test the compatibility of the latest protocol version with
+        // all previous protocol versions on testnet. It does this by starting a testnet with
+        // the old protocol version that this binary supports, and then upgrading it all the
+        // way to the latest protocol version.
+        let min_ver = ProtocolVersion::MIN.as_u64();
+        let max_ver = ProtocolVersion::MAX.as_u64();
+        let init_framework =
+            sui_framework_snapshot::load_bytecode_snapshot("testnet", min_ver).unwrap();
+        let mut test_cluster = init_test_cluster_builder(7, 5000)
+            .with_protocol_version(ProtocolVersion::MIN)
+            .with_supported_protocol_versions(SupportedProtocolVersions::new_for_testing(
+                min_ver, min_ver,
+            ))
+            .with_fullnode_supported_protocol_versions_config(
+                SupportedProtocolVersions::new_for_testing(min_ver, max_ver),
```

Comment:
Why do we want the fullnode to support all of the min..max versions?

Reply:
Because fullnode always follow the network, so there is no point to change what version range it supports in the middle of the test